### ISSUE TITLE
Fatal Error on stack rename

### ISSUE THREAD

**Issue by headconnect** [pulumi/pulumi#6526]
Attempted a stack rename using "fully qualified" stack name in order to attempt to change the project name of the stack. 
## Expected behavior

New stack name in New Project
## Current behavior

*Note:* only using local state on this, nothing remote, no pulumi.com. Aware that the whole organizational element is a pulumi.com thing, but I figured why not try.
By calling `pulumi stack rename StandardProject/Dev-Testing-Resources` I get a new stack inside the old project called "StandardProject-Dev-Testing-Resources".
By calling , pulumi stack rename 'ORG/StandardProject/Dev-Testing-Resources' I get: 
_(org/standardproject/dev-testing-resources has been renamed from original input for reasons of unnecessary confidentiality)_
```
 pulumi stack rename 'ORG/StandardProject/Dev-Testing-Resources'
================================================================================
The Pulumi CLI encountered a fatal error. This is a bug!
We would appreciate a report: https://github.com/pulumi/pulumi/issues/
Please provide all of the below text in your report.
================================================================================
Pulumi Version:   v2.22.0
Go Version:       go1.15.8
Go Compiler:      gc
Architecture:     amd64
Operating System: windows
Panic:            runtime error: slice bounds out of range [-1:]

goroutine 1 [running]:
runtime/debug.Stack(0xc0005d1650, 0x1e40260, 0xc00003b080)
        C:/hostedtoolcache/windows/go/1.15.8/x64/src/runtime/debug/stack.go:24 +0xa5
main.panicHandler()
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/cmd/pulumi/main.go:29 +0x75
panic(0x1e40260, 0xc00003b080)
        C:/hostedtoolcache/windows/go/1.15.8/x64/src/runtime/panic.go:969 +0x1c7
github.com/pulumi/pulumi/pkg/v2/backend/filestate.(*localBackend).renameHistory(0xc0006509b0, 0xc00016c180, 0x24, 0xc0000ca030, 0x2a, 0x3c, 0xc0005e5040)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/backend/filestate/state.go:406 +0x55a
github.com/pulumi/pulumi/pkg/v2/backend/filestate.(*localBackend).RenameStack(0xc0006509b0, 0x22932a0, 0xc0000a20a0, 0x22b12a0, 0xc0014d20f0, 0xc0000ca030, 0x2a, 0xc001468230, 0xc00015a8a0, 0xc001498280, ...)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/backend/filestate/backend.go:393 +0x24e
github.com/pulumi/pulumi/pkg/v2/backend.RenameStack(0x22932a0, 0xc0000a20a0, 0x22b12a0, 0xc0014d20f0, 0xc0000ca030, 0x2a, 0x1be15c0, 0xc0014d43a0, 0x1be15c0, 0xc0014d43b0)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/backend/stack.go:74 +0x9b
github.com/pulumi/pulumi/pkg/v2/backend/filestate.(*localStack).Rename(0xc0014d20f0, 0x22932a0, 0xc0000a20a0, 0xc0000ca030, 0x2a, 0x0, 0x0, 0x0, 0x0)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/backend/filestate/stack.go:64 +0x6d
main.newStackRenameCmd.func1(0xc0004a02c0, 0xc00033b640, 0x1, 0x1, 0xc, 0xc)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/cmd/pulumi/stack_rename.go:66 +0x1f8
github.com/pulumi/pulumi/sdk/v2/go/common/util/cmdutil.RunFunc.func1(0xc0004a02c0, 0xc00033b640, 0x1, 0x1, 0x0, 0x0)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/sdk/go/common/util/cmdutil/exit.go:96 +0x58
github.com/pulumi/pulumi/sdk/v2/go/common/util/cmdutil.RunResultFunc.func1(0xc0004a02c0, 0xc00033b640, 0x1, 0x1)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/sdk/go/common/util/cmdutil/exit.go:112 +0x72
github.com/spf13/cobra.(*Command).execute(0xc0004a02c0, 0xc00033b5f0, 0x1, 0x1, 0xc0004a02c0, 0xc00033b5f0)
        D:/a/pulumi/pulumi/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc00057c580, 0x15, 0x0, 0x0)
        D:/a/pulumi/pulumi/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        D:/a/pulumi/pulumi/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/cmd/pulumi/main.go:48 +0x54
```
Aaand my stack is seemingly gone.
## Steps to reproduce

I would very much rather not try at this point :D

## Context (Environment)

I've been developing this stack for a while in its own project, and at this point I would like to formalize it a bit more from its 'silly' project and name to one that can be shared with the rest of the organization.

**Comment by headconnect** [comment]
It seems the `.pulumi\stacks` folder gives an indication - there seems to be an `'ORG` folder with a `StandardProject` folder containing `Dev-Testing-Resources'.json` (and .json.attrs) files. Note the `'` in the filenames there.

**Comment by vinayakakv** [comment]
I have same issue while using AWS s3 backend. While creating stacks via `pulumi stack init`, it did not allow slashes in the name to follow `orgName/stack` format. Thus, I ended up creating `stack` and renaming it to `orgName/projectName/stack` via `pulumi stack rename`.

Now that I could not see any output in the `pulumi stack ls` despite `pulumi stack` showing outputs, I thought naming is an issue and wanted the stack to renamed to `orgName/stack` and encountered this issue.

Fortunately, in the S3 bucket, `.pulumi/stacks/orgName/stack` was created and I moved it back to `.pulumi/stacks/orgName/projectName/stack`. I'm still not sure whether the stack name is causing it not to appear it in `pulumi stack ls` (even with `--all` option)
